File: frontend/src/types/matching.ts
/**
 * Mirrors backend: app/models/matching.py, app/schemas/matching.py
 */
import type { User } from "./user";

export interface MatchCandidate {
  mentor: User;
  score: number;
  explanation: string;
}

export interface MatchResponse {
  mentee_id: string;
  top_matches: MatchCandidate[];
}

export interface MatchingRecord {
  id: string;
  mentor_id: string;
  mentee_id: string;
  score: number;
  explanation: string | null;
  created_at: string;
}

File: backend/app/schemas/matching.py
"""
Pydantic schemas for the matching engine.
"""
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.schemas.user import UserOut


class MatchCandidate(BaseModel):
    mentor: UserOut
    score: float
    explanation: str


class MatchResponse(BaseModel):
    mentee_id: str
    top_matches: list[MatchCandidate]


class MatchingOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    mentor_id: str
    mentee_id: str
    score: float
    explanation: str | None = None
    created_at: datetime
